test/homepageFeaturedImageLoadingTest.js

```diff
@@ -0,0 +1,73 @@
+const puppeteer = require("puppeteer");
+
+function parseArgs(args) {
+  const parsedArgs = {};
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+    if (arg.startsWith("--")) {
+      const keyValue = arg.split("=");
+      const key = keyValue[0].slice(2);
+      const value = keyValue.length > 1 ? keyValue[1] : true;
+      parsedArgs[key] = value;
+    }
+  }
+  return parsedArgs;
+}
+
+(async () => {
+  const args = process.argv.slice(2);
+  const parsedArgs = parseArgs(args);
+
+  const browser = await puppeteer.launch({ headless: true });
+  const page = await browser.newPage();
+
+  await page.setRequestInterception(true);
+
+  const loadedImageUrls = [];
+
+  // Intercept network requests
+  page.on("request", (request) => {
+    if (request.resourceType() === "image") {
+      console.log("Image request:", request.url());
+      loadedImageUrls.push(request.url());
+    }
+    request.continue();
+  });
+
+  let homepage = "https://qa-new-digitalcollections.nypl.org";
+
+  if (parsedArgs["prod"] === true) {
+    console.log("Condition is true");
+    homepage = "https://new-digitalcollections.nypl.org";
+  }
+
+  await page.goto(homepage);
+
+  const startTime = new Date();
+
+  const desiredImageUrls = loadedImageUrls.filter((url) =>
+    url.includes("900,900")
+  );
+
+  await page.waitForFunction(
+    (desiredImageUrls) => {
+      const images = document.querySelectorAll("img");
+      const loadedImageUrls = Array.from(images).map((img) => img.src);
+
+      return desiredImageUrls.every((url) => loadedImageUrls.includes(url));
+    },
+    {},
+    desiredImageUrls
+  );
+
+  const loadTime = new Date() - startTime;
+  console.log(
+    "Time taken for all desired images to appear:",
+    loadTime,
+    "milliseconds"
+  );
+
+  console.log("Loaded image URLs:", loadedImageUrls);
```

Comment:
if loadedImageUrls is empty should it still be printing the "Time taken for all desired images to appear" ?

Reply:
Great question! I think if https://github.com/NYPL/digital-collections/blob/e1e5d1e6cbee3f36dd8f49d33c7690a89af038ef/test/homepageFeaturedImageLoadingTest.js#L52 times out (after 30s), the script exits with 1 and we don't even get that far?

Reply:
i'm getting this, but can access qa-new-digitalcollections.nypl.org just fine? Is there something else I might be missing, it works fine with --prod ![Screenshot 2024-04-10 at 1 32 42 PM](https://github.com/NYPL/digital-collections/assets/73774046/2f75c27b-2ed0-433a-9cc0-57a682bb0e3f)

Reply:
Thanks, I can take a look

Reply:
Hmm, I ran it just now and it worked fine. I am on the VPN and accessinated to nypl-dams-dev. Are you at the library today?

Reply:
If you are, can you try clearing your cache and loading the qa page to see if the images are there?

Reply:
yes I am at the library. On qa page swimlanes images load right away, featured image loads slowly

Reply:
But definitely within 5-10s, not 30s

Reply:
I can take another look at this. In any event, you shouldn't be getting an empty list of Loaded image URLs. Can you please add your public IP on that network to the ticket linked in the PR writeup?

Reply:
Actually it looks like it only happens on ethernet, works fine on the other office networks

Reply:
Spawned: https://jira.nypl.org/browse/DR-2917